### PR TITLE
11945 iOS updated ISwrveCommon to remove private functions used only by internal libraries

### PR DESCRIPTION
**Pull request by adam-govan**
https://swrvedev.jira.com/browse/SWRVE-11945

@Sergio-Mira / @dominicmarmionswrve  / @seaders

### CODE REVIEW

File: SwrveSDKCommon/Common/SwrveCommon.h [hunk 1]
```diff
@@ -3,11 +3,10 @@
 /*! Swrve SDK shared protocol (interface) definition */
 @protocol ISwrveCommon <NSObject>
 
+@required
```
Comment (Sergio-Mira) [inline SwrveSDKCommon/Common/SwrveCommon.h:6]: :+1:

File: SwrveSDKCommon/Common/SwrveCommon.h [hunk 2]
```diff
@@ -3,11 +3,10 @@
 /*! Swrve SDK shared protocol (interface) definition */
 @protocol ISwrveCommon <NSObject>
```
Comment (Sergio-Mira) [inline SwrveSDKCommon/Common/SwrveCommon.h]: We should change this protocol name too?

File: SwrveSDKCommon/Common/SwrveCommon.h [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
 -(BOOL) processPermissionRequest:(NSString*)action;
 
 @end
 
 @interface SwrveCommon : NSObject
 
-+(id<ISwrveCommon>) getSwrveCommon;
-+(void) setSwrveCommon:(id<ISwrveCommon>)swrveCommon;
++(id<SwrveCommonDelegate>) getSwrveCommon;
++(void) setSwrveCommon:(id<SwrveCommonDelegate>)swrveCommon;
```
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.h]: This is essentially a sharedInstance so shouldn't we do something similar to what we do in the main Swrve class:
https://github.com/Swrve/swrve-ios-sdk/blob/release-4_2/Sdk/Track/Swrve.m#L568
[SwrveCommon sharedInstance]
[SwrveCommon addSharedInstance:......]

**Comment by Sergio-Mira** [comment]
:rocket:

File: SwrveSDKCommon/Common/SwrveCommon.h [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
 -(BOOL) processPermissionRequest:(NSString*)action;
 
 @end
 
 @interface SwrveCommon : NSObject
 
-+(id<ISwrveCommon>) getSwrveCommon;
-+(void) setSwrveCommon:(id<ISwrveCommon>)swrveCommon;
++(id<SwrveCommonDelegate>) sharedInstance;
++(void) addSharedInstance:(id<SwrveCommonDelegate>)swrveCommon;
```
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.h:17]: :+1:

File: SwrveSDKCommon/Common/SwrveCommon.m [hunk 5]
```diff
@@ -1,17 +1,17 @@
 #import "SwrveCommon.h"
 
-static id<ISwrveCommon> _iSwrveCommon = NULL;
+static id<SwrveCommonDelegate> _sharedInstance = NULL;
```
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.m:3]: :+1:

**Pull request merged**